subt_ign/src/GameLogicPlugin.cc

```diff
@@ -414,12 +426,46 @@ void GameLogicPlugin::Configure(const ignition::gazebo::Entity & /*_entity*/,
   this->dataPtr->publishThread.reset(new std::thread(
         &GameLogicPluginPrivate::PublishScore, this->dataPtr.get()));
 
+  this->dataPtr->node.Subscribe("/subt_performer_detector",
+      &GameLogicPluginPrivate::OnEvent, this->dataPtr.get());
+
   ignmsg << "Starting SubT" << std::endl;
 
   // Make sure that there are score files.
   this->dataPtr->UpdateScoreFiles();
 }
 
+//////////////////////////////////////////////////
+void GameLogicPluginPrivate::OnEvent(const ignition::msgs::Pose &_msg)
+{
+  std::string frameId = "nil";
+  std::string state = "nil";
+
+  for (int i = 0; i < _msg.header().data_size(); ++i)
+  {
+    if (_msg.header().data(i).key() == "frame_id")
+      frameId = _msg.header().data(i).value(0);
+    else if (_msg.header().data(i).key() == "state")
+    {
+      if (_msg.header().data(i).value(0) == "1")
+        state = "enter";
+      else
+        state = "exit";
+    }
+  }
+
+  this->eventStream
+    << "- event:\n"
+    << "    time_sec: " << _msg.header().stamp().sec() << "\n"
+    << "    detector: " << frameId << "\n"
+    << "    robot: " << _msg.name() << "\n"
+    << "    state: " << state << "\n"
+    << "    pos:\n"
```

Comment:
The position published by the performer_detector system is relative to the pose of the model containing the system. Will that be an issue?

Reply:
Ahh, thanks. I was printing it because the information was there. It's not important to have that bit. Removed in: 7888f4fa1995f01c49c51816be4b95b11c3811cf